sqlmesh/core/model/common.py

```diff
@@ -19,7 +19,8 @@ def parse_model_name(name: str) -> t.Tuple[t.Optional[str], t.Optional[str], str
     Returns:
         A tuple consisting of catalog, schema, table name.
     """
-    return split_num_words(name, ".", 3)  # type: ignore
+    table = exp.to_table(name)
+    return table.catalog or None, table.db or None, table.name
```

Comment:
I was having an issue with this after fixing the forward-only for external models. Seems like it is time for us to move away from string splitting to get this information. I wasn't sure about dialect here with recent changes related to dialect.

Reply:
right i forgot to remove this with my refactor